resolwe_bio/processes/support_processors/prepare_geo.yml

```diff
@@ -0,0 +1,233 @@
+# ===========
+# Prepare GEO
+# ===========
+---
+
+- slug: prepare-geo-chipseq
+  name: Prepare GEO - ChIP-Seq
+  requirements:
+    expression-engine: jinja
+    resources:
+      cores: 1
+    executor:
+      docker:
+        image: resolwebio/base:ubuntu-16.04
+  data_name: 'Prepare GEO ({{name}} - ChIP-Seq)'
+  version: 0.0.1
+  type: data:other:geo:chipseq
+  category: analyses
+  persistence: CACHED
+  description: >
+    Prepare ChIP-Seq data for GEO upload.
+  input:
+    - name: reads
+      label: Reads
+      type: list:data:reads:fastq
+      description: >
+        List of reads objects. Fastq files will be used.
+    - name: macs14
+      label: Macs14
+      type: list:data:chipseq:macs14
+      description: >
+        List of macs14 objects. Wiggle files will be used.
+    - name: relations
+      label: Relations
+      type: list:basic:string
+      description: >
+        Sample to background mapper. Accepted format <sample name>:<background sample name>.
+    - name: name
+      label: Collection name
+      type: basic:string
+  output:
+    - name: tarball
+      label: GEO folder
+      type: basic:file
+    - name: table
+      label: Annotation table
+      type: basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      NAME={{ name }}_ChIP-Seq
+      mkdir ${NAME}
+
+      declare -A SAMPLE_TO_PAIRED
+      declare -A SAMPLE_TO_BACKGROUND
+
+      {% for relation in relations %}
+        SAMPLE_TO_BACKGROUND[{{ relation.split(':')[0].split('.')[0] }}]={{ relation.split(':')[1].split('.')[0] }}
+      {% endfor %}
+
+      echo -e "File_name\tSample_name\tmd5sum\tSE/PE\tRead_length" > "${NAME}".txt
+
+      {% for read in reads %}
+        SAMPLE_NAME=`basename {{ read|sample_name }} .fastq.gz`
+        PAIRED=SE
+        MATE1_LABEL=""
+        MATE2_LABEL=""
+        {% if read|type|subtype('data:reads:fastq:paired:') %}
+          PAIRED=PE
+          MATE1_LABEL=_R1
+          MATE2_LABEL=_R2
+        {% endif %}
+        SAMPLE_TO_PAIRED["${SAMPLE_NAME}"]=$PAIRED
+
+        {% for fastq in read.fastq %}
+          cp {{ fastq.file }} "${NAME}/${SAMPLE_NAME}${MATE1_LABEL}.fastq.gz"
+          MD5=($(md5sum "${NAME}/${SAMPLE_NAME}${MATE1_LABEL}.fastq.gz"))
+
+          zcat "${NAME}/${SAMPLE_NAME}${MATE1_LABEL}.fastq.gz" |
+          head -n 4000 |
+          awk '(NR + 2) % 4 == 0' > subsample.txt
+
+          LENGTH=`reads_length.py subsample.txt`
```

Comment:
LENGTH=`awk '{print length}' subsample.txt | sort | uniq -c | awk -F ' ' '{print $2}' | sort -r | head -n1`
But add a commentary what this does.